### PR TITLE
Fix several issues around catch beatmap conversion

### LINKED ISSUE #15305: Hyperdash generation difference between lazer and stable can be a big difficulty change
For example, in [this map](https://osu.ppy.sh/beatmapsets/1541421#fruits/3150892), the the fruit at 196x combo is a hyper fruit in lazer but is a normal fruit in stable.
This is due to stable truncating time variables to integers in random places.
Note that `CatchBeatmapConversionTest` doesn't catch this difference because hyper fruit information is not presented in the json.

I guess some difference from stable is expected, but the osu!catch-specific issue here is that whether a fruit is hyper or not can be a big difficulty difference, especially in converted beatmaps.
In osu!catch convert maps, the main difficulty source of a map is "pixel jumps" where the pattern is not wide enough to become a hyper-dash, but just barely below the normal dashing distance.
It is not rare to have a situation where one pixel-jump pattern transforms an easy map to an "unplayable" very difficult map.

The effect on the star rating/pp shouldn't be large currently because the difficulty of such pixel jump patterns is not accurately measured, but if the difficulty calculation algorithm is improved, the impact can be large.

Should we try to reduce the difference between lazer and stable's hyper-dash generation?

### PR DESCRIPTION
**Pull request by smoogipoo**
Fixes https://github.com/ppy/osu/issues/15305

This was originally intended to fix hyperdash generation around "pixeljumps", however it turned into much more general fixing of beatmap conversion because I couldn't provide beatmap conversion comparisons without fixing other issues.

This PR makes lazer match stable in:
- Pixeljump hyperdashes
- Number of tiny ticks
- Number of bananas

Spreadsheet of differences: https://docs.google.com/spreadsheets/d/1nkmmdLIob8oD6vGjOirpz4l6SniwHaV-6P23sm_pHp8/edit

I've provided tests for each of the individual fixed cases (`pixel-jump`, `tiny-ticks`, `v8-tick-distance`, `spinner-precision`), as well as whole beatmaps from some rows of the sheet above.

### CODE REVIEW

**Comment by peppy** [comment]
Holding off reviewing this pending some incoming changes.

File: osu.Game.Rulesets.Catch/Objects/JuiceStream.cs [hunk 1]
```diff
@@ -41,6 +41,12 @@ public double SliderVelocityMultiplier
             set => SliderVelocityMultiplierBindable.Value = value;
         }
 
+        /// <summary>
+        /// An extra multiplier that affects the number of <see cref="Droplet"/>s generated by this <see cref="JuiceStream"/>.
+        /// An increase in this value increases <see cref="TickDistance"/>, which reduces the number of ticks generated.
+        /// </summary>
+        public double TickDistanceMultiplier = 1;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/Objects/JuiceStream.cs:48]: Is it worth noting in the xmldoc that this is used for legacy purposes? Or are you thinking this something we might use in the future?

File: osu.Game.Rulesets.Catch/Objects/BananaShower.cs [hunk 2]
```diff
@@ -23,29 +23,29 @@ protected override void CreateNestedHitObjects(CancellationToken cancellationTok
 
         private void createBananas(CancellationToken cancellationToken)
         {
-            double spacing = Duration;
+            int startTime = (int)StartTime;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/Objects/BananaShower.cs:26]: Probably inline a comment here mentioning the flooring is for stable compatibility.

File: osu.Game.Rulesets.Catch/Objects/JuiceStream.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Catch/Objects/JuiceStream.cs:48]: I copied this xmldoc straight from the osu! ruleset.

Comment (peppy) [reply osu.Game.Rulesets.Catch/Objects/JuiceStream.cs:48]: Hmm okay.

**Review by peppy — APPROVED** [review]

**Pull request merged**